router/src/lib.rs

```diff
@@ -956,48 +966,72 @@ pub fn default_tool_prompt() -> String {
     "\nGiven the functions available, please respond with a JSON for a function call with its proper arguments that best answers the given prompt. Respond in the format {name: function name, parameters: dictionary of argument name and its value}.Do not use variables.\n".to_string()
 }
 
-#[derive(Clone, Debug, Deserialize, PartialEq, Serialize, ToSchema)]
-#[schema(example = "auto")]
-/// Controls which (if any) tool is called by the model.
-pub enum ToolType {
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[serde(tag = "type")]
+pub enum TypedChoice {
+    #[serde(rename = "function")]
+    Function { function: FunctionName },
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct FunctionName {
+    pub name: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, ToSchema, Default)]
+#[serde(from = "ToolTypeDeserializer")]
+pub enum ChatCompletionToolChoiceOption {
     /// Means the model can pick between generating a message or calling one or more tools.
     #[schema(rename = "auto")]
-    OneOf,
+    Auto,
     /// Means the model will not call any tool and instead generates a message.
     #[schema(rename = "none")]
+    #[default]
     NoTool,
+    /// Means the model must call one or more tools.
+    #[schema(rename = "required")]
+    Required,
     /// Forces the model to call a specific tool.
     #[schema(rename = "function")]
+    #[serde(alias = "function")]
```

Comment:
Can we have link in rustdoc to where all these attributes are explained, possibly in the original openai spec (or anthropic)

Reply:
yep, just added a useful link above the enum that explains all 4 of the options in OpenAI's docs this link will also be included in the openapi schema for the `ChatCompletionToolChoiceOption` description